internal/gengapic/testdata/empty_client_init.want

```diff
@@ -1,4 +1,16 @@
-// Client is a client for interacting with Awesome Foo API.
+// internalClient is an interface that defines the methods availaible from Awesome Foo API.
+type internalClient interface {
+	Zip(context.Context, *mypackagepb.Bar, ...gax.CallOption) (*mypackagepb.Foo, error)
+}
+
+//  is a thin wrapper that holds an internalClient
+// It is agnostic as to the underlying transport, i.e. json+http, gRPC, or other.
+type  struct {
```

Comment:
Need to handle case where `servName` is empty (the resulting client struct name is just "Client")

Reply:
Can you explain again why we want this struct?

Reply:
We want to provide some means of polymorphic abstraction over grpc/rest+json. In the last review, Cody pointed out that adding methods to a public interface is a breaking change, even though the interface and the two implementing structs are generated concurrently and therefore guaranteed to be in sync.
The workaround is to make the interface private and make the public type a struct. Adding methods to a struct is not breaking, and the struct just embeds the interface.

To use the monikers from Python, the existing client is a 'transport', for which there exists an ~abstract base class~ interface. The new wrapper struct is the Python 'client' equivalent.

Reply:
Changed the naming convention away from `p("%s", serv.name)` to `p("%sClientHandle", serv.name)`

Reply:
Ok let me see if I understand this. A new factory method like `NewFooClientHandle () FooClientHandle` where `FooClientHandle` is 
```go
type FooClientHandle struct {
  internalFooInterface
}
```

`NewFooClientHandle` would return a `FooClientHandle` backed by either the `FooRestClient` or `FooClient` (grpc). Is that right?

_Ignore naming and stuff for now, that is easily rectified._

Reply:
Exactly. The factory function signature(s) and implementation (s) are the subject of a subsequent PR.